build-tools-internal/src/main/java/org/elasticsearch/gradle/internal/test/rest/transform/headers/InjectHeaders.java

```diff
@@ -39,6 +40,10 @@ public InjectHeaders(Map<String, String> headers) {
     @Override
     public void transformTest(ObjectNode doNodeParent) {
         ObjectNode doNodeValue = (ObjectNode) doNodeParent.get(getKeyToFind());
+        if(isCatOperation(doNodeValue)){
```

Comment:
Rather than making this class aware of "cat." names, can you introduce conditional headers to inject ?

So something like : 
```
public InjectHeaders(Map<String, String> headers, Map<Function<ObjectNode, Boolean>, Map<String, String>> conditionalHeaders)
```

Then if the function key evaluates to true, then insert the associated headers.

Reply:
@jakelandis good idea, but I think this should be just set of functions and when they all evaluate to true, then we would insert the headers.
implemented this here https://github.com/elastic/elasticsearch/pull/74701/commits/7ad7093637609fc420bf04e1483c6320788dabd6 - let me know what you think

Reply:
I created a separate PR for this change 
https://github.com/elastic/elasticsearch/pull/75001

Reply:
:heavy_check_mark: over at #75001 , thanks for the iterations.